File: app/src/main/java/org/cuieney/videolife/di/component/ActivityComponent.java
package org.cuieney.videolife.di.component;

import android.app.Activity;


import org.cuieney.videolife.di.PerActivity;
import org.cuieney.videolife.di.module.ActivityModule;

import dagger.Component;

/**
 * Created by cuieney on 16/8/7.
 */

@PerActivity
@Component(dependencies = AppComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    Activity getActivity();


}

File: app/src/main/java/org/cuieney/videolife/kotlin/di/component/ActivityComponent.kt
package org.cuieney.videolife.kotlin.di.component

import android.app.Activity
import dagger.Component
import org.cuieney.videolife.di.PerActivity
import org.cuieney.videolife.kotlin.di.module.ActivityModule
import org.cuieney.videolife.kotlin.ui.act.VrActivity

/**
 * Created by cuieney on 2017/5/22.
 */
@PerActivity
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent{
    fun getActivity(): Activity
    fun inject(vrActivity: VrActivity)
}